backend/src/models/entry/entryAnalysis.ts

```diff
@@ -74,32 +77,49 @@ entryAnalysisSchema.methods.getAnalysisContent = async function (configId: strin
       }
     }
   }
+  return config.model.analysis;
+}
 
-  const cdGpt = new CdGpt(process.env.OPENAI_API_KEY, config.model.analysis);
+async function getAnalysisCompletion(configModelAnalysis: string, content: string): Promise<any> {
+  if (process.env.OPENAI_API_KEY === undefined) {
+    throw new Error('OPENAI_API_KEY undefined. Must be set to get analysis')
+  }
+  const cdGpt = new CdGpt(process.env.OPENAI_API_KEY, configModelAnalysis, '', .7);
 
   cdGpt.seedAnalysisMessages();
   cdGpt.addUserMessage({ analysis: content });
 
-  const analysisCompletion = await cdGpt.getAnalysisCompletion();
+  const analysisCompletion: any = await cdGpt.getAnalysisCompletion();
 
   if (analysisCompletion.error) {
     throw new Error(analysisCompletion.error.message);
   }
 
-  const response = JSON.parse(analysisCompletion.choices[0].message.content);
+  return JSON.parse(analysisCompletion.choices[0].message.content);
+}
 
-  const { reframed_thought: reframing, distortion_analysis: analysis, impact_assessment: impact, affirmation, is_healthy: isHealthy } = response;
+// Get the analysis content for an entry
+entryAnalysisSchema.methods.getAnalysisContent = async function (configId: string, content: string): Promise<any> {
```

Comment:
I like the refactor. Cleans things up nicely!

Reply:
Then I'll be doing the same in entryCovnersation too